### PR TITLE
Temporary solution to duplicate Exchange folders

### PR DESCRIPTION
**Pull request by ashmrtn**
For Exchange email or contacts backup, pick on the folder with
the highest ID in the hopes that it will be the most recently
used one

A better solution for this situation should be coming soon

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [x] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #3197

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/exchange/service_iterators.go [hunk 1]
```diff
@@ -99,6 +103,52 @@ func filterContainersAndFillCollections(
...
+		// should be skipped or the collection we previously added should be
+		// skipped.
+		//
+		// Calendars is already using folder IDs so we don't need to pick the
+		// "newest" folder for that.
+		if oldCID := dupPaths[locPath.String()]; category != path.EventsCategory && len(oldCID) > 0 {
+			if cID < oldCID {
+				logger.Ctx(ictx).Infow(
+					"skipping duplicate folder with lesser ID",
+					"previous_folder_id", clues.Hide(oldCID),
```
Comment (vkamra) [inline src/internal/connector/exchange/service_iterators.go:116]: Let's log the path as well (here and message below)

File: src/internal/connector/exchange/service_iterators.go [hunk 2]
```diff
@@ -51,6 +51,10 @@ func filterContainersAndFillCollections(
 		// deleted from this map, leaving only the deleted folders behind
 		tombstones = makeTombstones(dps)
 		category   = qp.Category
+
+		// Stop-gap: Track folders by LocationPath and if there's duplicates pick
+		// the one with the lexicographically larger ID.
```
Comment (pandeyabs) [inline src/internal/connector/exchange/service_iterators.go:56]: Curious, is this documented behavior that larger ID == more recently created?

**Review by vkamra — APPROVED** [review]
Reviewed test cases offline with @ashmrtn . LGTM.

@ashmrtn - will address logging comment prior to merge.

File: src/internal/connector/exchange/service_iterators.go (hunk 2 above)
Comment (ashmrtn) [reply src/internal/connector/exchange/service_iterators.go:56]: This isn't documented anywhere, just something we've noticed as we've gone along. We're not hoping to rely on it in the long run, this is more a temporary thing to unblock some folks until we have a better solution ready

Comment (pandeyabs) [reply src/internal/connector/exchange/service_iterators.go:56]: Gotcha! Better solution is moving to folder ids?

**Review by pandeyabs — COMMENTED** [review]
The logic looks good to me.

**Pull request merged**